cmd/api/src/daemons/datapipe/jobs.go

```diff
@@ -51,12 +51,16 @@ func (s *Daemon) completeJobsUnderAnalysis() {
 }
 
 func (s *Daemon) processIngestedFileUploadJobs() {
-	if ingestedFileUploadJobs, err := s.db.GetFileUploadJobsWithStatus(model.JobStatusIngesting); err != nil {
+	if ingestingFileUploadJobs, err := s.db.GetFileUploadJobsWithStatus(model.JobStatusIngesting); err != nil {
 		log.Errorf("Failed to look up finished file upload jobs: %v", err)
 	} else {
-		for _, ingestedFileUploadJob := range ingestedFileUploadJobs {
-			if err := fileupload.UpdateFileUploadJobStatus(s.db, ingestedFileUploadJob, model.JobStatusAnalyzing, "Analyzing"); err != nil {
-				log.Errorf("Error updating fileupload job %d: %v", ingestedFileUploadJob.ID, err)
+		for _, ingestingFileUploadJob := range ingestingFileUploadJobs {
+			if remainingIngestTasks, err := s.db.GetIngestTasksForJob(ingestingFileUploadJob.ID); err != nil {
+				log.Errorf("Failed looking up remaining ingest tasks for file upload job %d: %v", ingestingFileUploadJob.ID, err)
+			} else if len(remainingIngestTasks) == 0 {
+				if err := fileupload.UpdateFileUploadJobStatus(s.db, ingestingFileUploadJob, model.JobStatusAnalyzing, "Analyzing"); err != nil {
+					log.Errorf("Error updating fileupload job %d: %v", ingestingFileUploadJob.ID, err)
+				}
```

Comment:
Not sure how difficult it would be to write an integration test for this but it would be nice to prevent regressions. Feel free to make that call, I'll go ahead and stamp this

Reply:
I don't actually think this needs an integration test. I need to think about this for a moment but let me see what I can come up with. The state transitions should be relatively easy to mock out.

Thanks for calling this out!